File: webapp/src/app/shared/models/auth/auth-dto.ts
export interface LoginRequest {
  email: string,
  password: string
}

export interface TokenRequest {
  refreshToken?: string
}

export interface LoginResponse {
  refreshToken?: string,
  accessToken?: string,
}

export interface TokenResponse {
  accessToken?: string
}

export interface Permissions {
  id?: number
  name?: string,
}

File: src/main/kotlin/com/exampledigisphere/vitapetcare/auth/DTO/AuthDTO.kt
package com.exampledigisphere.vitapetcare.auth.DTO

import jakarta.validation.constraints.NotBlank

data class LoginRequest(
  @field:NotBlank val email: String,
  @field:NotBlank val password: String
)

data class TokenRequest(
  @field:NotBlank val refreshToken: String,
)

data class LoginResponse(
  val refreshToken: String,
  val accessToken: String
)

data class TokenResponse(
  val accessToken: String
)

data class PermissionsResponse(
  val id: Int,
  val name: String,
)
